pkg/providers/vsphere/virtualmachine/cdrom.go

```diff
@@ -139,12 +162,13 @@ func updateCdromDeviceChangesWithSharedDisks(
 	)
 	deviceChanges = append(deviceChanges, curCdromChanges...)
 
-	return deviceChanges, nil
+	configSpec.DeviceChange = append(configSpec.DeviceChange, deviceChanges...)
+	return nil
 }
 
-// updateCdromDeviceChangesLegacy implements CD-ROM reconciliation with
+// UpdateCdromDeviceChangesLegacy implements CD-ROM reconciliation with
 // automatic controller assignment (VMSharedDisks disabled).
```

Comment:
Maybe also add, it creates controllers on demand.

Reply:
sounds good. will do